### PR TITLE
Gtm matm optimizations

### PR DESCRIPTION
**Pull request by maxsam4**
Ported changes from #601

### CODE REVIEW

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol [hunk 1]
```diff
@@ -260,57 +264,57 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
...
+    * @param _canSendAfter is the moment when the sale lockup period ends and the investor can freely sell his tokens
+    * @param _canReceiveAfter is the moment when the purchase lockup period ends and the investor can freely purchase tokens from others
     * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
     */
     function modifyKYCDataMulti(
         address[] memory _investors,
-        uint256[] memory _fromTime,
-        uint256[] memory _toTime,
-        uint256[] memory _expiryTime
+        uint64[] memory _canSendAfter,
```
Comment (satyamakgec) [inline contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:303]: in 2.1.0 you used `uint256` in function arguments and then typecast those into uint64 in the function body. Shouldn't we make consistency in both branches

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol [hunk 2]
```diff
@@ -18,19 +18,19 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
...
+    // if allowAllWhitelistIssuances is TRUE, then _canReceiveAfter is ignored when receiving tokens from the issuance address
+    // if allowAllWhitelistTransfers is TRUE, then _canReceiveAfter and _canSendAfter is ignored when sending or receiving tokens
     // in any case, any investor sending or receiving tokens, must have a _expiryTime in the future
     event ModifyKYCData(
         address indexed _investor,
         address indexed _addedBy,
-        uint256 _fromTime,
-        uint256 _toTime,
-        uint256 _expiryTime
+        uint64 _canSendAfter,
```
Comment (satyamakgec) [inline contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:31]: Shouldn't it uint256 for consistency with dev-2.1.0

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol [hunk 3]
```diff
@@ -260,57 +264,57 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
...
+    * @param _canReceiveAfter is the moment when the purchase lockup period ends and the investor can freely purchase or receive tokens from others
     * @param _expiryTime is the moment till investors KYC will be validated. After that investor need to do re-KYC
     */
     function modifyKYCData(
         address _investor,
-        uint256 _fromTime,
-        uint256 _toTime,
-        uint256 _expiryTime
+        uint64 _canSendAfter,
+        uint64 _canReceiveAfter,
```
Comment (satyamakgec) [inline contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:274]: should be uint256

**Review by satyamakgec — CHANGES_REQUESTED** [review]

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol (hunk 1 above)
Comment (maxsam4) [reply contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:303]: We used `uint256` in 2.1 to make the interface compatible with 2.0. (We want to use the same dApp on 2.0 and 2.1 version of security token/modules)

There is no need to keep 3.0 and 2.1 compatible as they will require different dApps anyway.

**Review by satyamakgec — APPROVED** [review]

**Pull request merged**